components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -185,6 +195,13 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
                     if (log.isDebugEnabled()) {
                         log.debug("Session data key is null in the request and not a logout request.");
                     }
+                    if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                        LoggerUtils.triggerDiagnosticLogEvent(
+                                FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK, null, LogConstants.FAILED,
+                                "Not a logout request and session data key " +
+                                        "is null in the request. Sending to retry page.", "handle-authentication" +
```

Comment:
better to reword a bit here as well. also lets have the exact param name "sessionDataKey" so that they can debug easily. And better to say sdk is "not provided or is null in the request"

Reply:
*session data key parameter (add the request parameter as it is) is not provided in the request.

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/4326/commits/ca34cc4be2ac20f9c870f1ded78fab17d9bf6d94